django/db/models/sql/compiler.py

```diff
@@ -830,9 +830,12 @@ def execute_sql(self, result_type=MULTI, chunked_fetch=False):
         try:
             cursor.execute(sql, params)
         except Exception:
-            with self.connection.wrap_database_errors:
-                # Closing a server-side cursor could yield an error
+            try:
+                # Might fail for server-side cursors (eg: connection closed)
                 cursor.close()
+            except Exception:
+                # Ignore clean-up errors. Raise the original error instead.
+                pass
```

Comment:
Hmm I should have read that a little better. Looks like six.raise_from raises the inner exception on python 2 which is what you're trying to avoid. Maybe a comment mentioning a change for when we drop py2, and a listing in the py3 ticket?